### ISSUE TITLE
High memory usage with IndexedSlices

### ISSUE THREAD

**Issue by thedomdom** [tensorflow/privacy#126]
Hi,

I'm using an Embedding layer in my model. 
When I train it with `DPAdamOptimizer`, I get a warning: 

`UserWarning: Converting sparse IndexedSlices to a dense Tensor with 163617900 elements. This may consume a large amount of memory.
`

Therefore, I have to use hardware with a huge amount of memory.
Is this intended behavior (as also 0-gradients have to be perturbed) or could this possibly be improved?
